apidef/oas/middleware.go

```diff
@@ -51,21 +51,37 @@ type Global struct {
 	CORS *CORS `bson:"cors,omitempty" json:"cors,omitempty"`
 
 	// PrePlugin contains configuration related to the custom plugin that is run before authentication.
-	// Tyk classic API definition: `custom_middleware.pre`.
+	// Deprecated: Use PrePlugins instead.
 	PrePlugin *PrePlugin `bson:"prePlugin,omitempty" json:"prePlugin,omitempty"`
 
+	// PrePlugins contains configuration related to the custom plugin that is run before authentication.
+	// Tyk classic API definition: `custom_middleware.pre`.
+	PrePlugins CustomPlugins `bson:"prePlugins,omitempty" json:"prePlugins,omitempty"`
```

Comment:
Keep it a pointer, it keeps allocations down whenever plugins are not configured (== nil).

Reply:
Do we need pointer since `CustomPlugins` is a slice
```
type CustomPlugins []CustomPlugin
```